cdap-docs/build.sh

```diff
@@ -38,227 +66,197 @@ function set_project_path() {
+  echo "========================================================"
+  echo "========================================================"
   echo ""
-
-  for i in ${MANUALS}; do
-    copy_html ${i}
-  done
-
-  local project_dir
-  # Rewrite 404 file, using branch if not a release
-  if [ "x${GIT_BRANCH_TYPE}" == "xfeature" ]; then
-    project_dir=${PROJECT_VERSION}-${GIT_BRANCH}
-  else
-    project_dir=${PROJECT_VERSION}
-  fi
-  rewrite ${BUILD}/${HTML}/404.html "src=\"_static"  "src=\"/cdap/${project_dir}/en/_static"
-  rewrite ${BUILD}/${HTML}/404.html "src=\"_images"  "src=\"/cdap/${project_dir}/en/_images"
-  rewrite ${BUILD}/${HTML}/404.html "/href=\"http/!s|href=\"|href=\"/cdap/${project_dir}/en/|g"
-  rewrite ${BUILD}/${HTML}/404.html "action=\"search.html"  "action=\"/cdap/${project_dir}/en/search.html"
+  exit 0
 }
 
-################################################## current
-
-function build_all() {
-  echo ""
-  echo "========================================================"
+function build_javadocs() {
   echo "========================================================"
-  echo "Building All: GitHub and Web Docs."
+  echo "Building Javadocs"
   echo "--------------------------------------------------------"
   echo ""
-  echo "========================================================"
-  echo "Building GitHub Docs."
+  build_javadocs_api
   echo "--------------------------------------------------------"
-  echo ""
-  run_command docs-github-part ${ARG_2} ${ARG_3}
-  echo "Stashing GitHub Docs."
-  cd ${SCRIPT_PATH}
-  mkdir -p ${SCRIPT_PATH}/${BUILD_TEMP}
-  mv ${SCRIPT_PATH}/${BUILD}/*.zip ${SCRIPT_PATH}/${BUILD_TEMP}
-  echo ""
+  echo "Completed Build of Javadocs"
   echo "========================================================"
-  echo "Building Web Docs."
-  echo "--------------------------------------------------------"
-  echo ""
-  run_command docs-web-part ${ARG_2} ${ARG_3}
   echo ""
+}
+
+function build_javadocs_api() {
+  set_mvn_environment
+  MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn clean install -P examples,templates,release -DskipTests -Dgpg.skip=true && mvn clean site -DskipTests -P templates -DisOffline=false
+}
+
+function build_docs_first_pass() {
   echo "========================================================"
-  echo "Replacing GitHub Docs."
+  echo "Building First Pass of Docs"
   echo "--------------------------------------------------------"
   echo ""
-  mv ${SCRIPT_PATH}/${BUILD_TEMP}/*.zip ${SCRIPT_PATH}/${BUILD}
-  rm -rf ${SCRIPT_PATH}/${BUILD_TEMP}
-  echo ""
+  build_docs_inner_level "docs"
   echo "--------------------------------------------------------"
-  bell "Completed \"build_all\"."
-  echo "========================================================"
+  echo "Completed Building First Pass of Docs"
   echo "========================================================"
   echo ""
-  exit 0
-}
-
-function build_javadocs() {
-  # Uses function from common
-  build_javadocs_api
+  return $?
```

Comment:
This will return 0, always. The previous command is `echo`... maybe you meant to save the return value of a previous command and return with that?